lib/index.js

```diff
@@ -39,30 +39,43 @@ module.exports = function setupDevEnv({
     const initAllChaincode = () => Promise.all(chaincodeLocations.map(location => initChaincode(location)));
     const installAllNpmPackages = () => chaincodeLocations.reduce((p, location) => p.then(() => installNpmPackages(location)), Promise.resolve());
 
-    // Upgrade chaincode whenever source files change
-    chaincodeLocations.forEach((location, index) => {
-        let version = 0;
-        chokidar.watch([
-            path.join(location, '**/*.js'),
-            `!${path.join(location, 'node_modules/**')}`
-        ], {ignoreInitial: true})
-            .on('all', (event, path) => {
-                console.log('Upgrading chaincode for ', location);
-                // Increase version number
-                version += 1;
-                // TODO: only copy changed file
-                copyChaincode(location).then(() => {
-                    upgradeChaincode(location, version)
-                        .then(() => console.log('Chaincode upgraded'))
-                        .catch((err) => console.log(err));
-                }).catch(err => console.log(err));
-            });
-    });
-
     // Start docker
     startShell(path.join(SCRIPTS_PATH, 'setupDocker.sh'), {}, DOCKER_SETUP_FINISHED_REGEX, 30000)
         .then(copyAllChaincode)
         .then(installAllNpmPackages)
         .then(initAllChaincode)
+        .then(() => {
+            // start watching
+
+            // Upgrade chaincode whenever source files change
+            chaincodeLocations.forEach((location, index) => {
+                let upgradePromise = Promise.resolve();
```

Comment:
Maybe we should put a comment here why we are chaining the upgrade step here rather then doing it in parallell.